src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/metrics/booleanexpressioncomplexity/Example1.java

```diff
@@ -0,0 +1,27 @@
+/*xml
+<module name="Checker">
+  <module name="TreeWalker">
+    <module name="BooleanExpressionComplexity"/>
+  </module>
+</module>
+*/
+package com.puppycrawl.tools.checkstyle.checks.metrics.booleanexpressioncomplexity;
+
+// xdoc section -- start
+public class Example1
+{
+  public static void main(String ... args)
+  {
+    boolean a = true;
+    boolean b = false;
+
+    boolean c = (a & b) | (b ^ a); // OK, 1(&) + 1(|) + 1(^) = 3 (max allowed 3)
+
+    boolean d = (a & b) | (b ^ a) | (a ^ b); // violation
+    // violation above, 1(&) + 1(|) + 1(^) + 1(|) + 1(^) = 5
```

Comment:
if you have `// violation above` you do not need `// violation`, please remove `// violation`

apply to all your PRs and changes in this PR.

Reply:
@romani ,
I did try to remove `// violation` a lot of times and the expected violations list always comes out empty : 

```
[ERROR] Failures:                                                                                                                     
[ERROR]   BooleanExpressionComplexityCheckExamplesTest.testExample1:42->AbstractModuleTestSupport.verifyWithInlineConfigParser:238->Ab
stractModuleTestSupport.verifyViolations:470 Violation lines for E:\MANISH\Workspace\checkstyle\src\xdocs-examples\resources\com\puppy
crawl\tools\checkstyle\checks\metrics\booleanexpressioncomplexity\Example1.java differ.
expected to be empty
but was: [20, 23]
[ERROR]   BooleanExpressionComplexityCheckExamplesTest.testExample2:51->AbstractModuleTestSupport.verifyWithInlineConfigParser:238->Ab
stractModuleTestSupport.verifyViolations:470 Violation lines for E:\MANISH\Workspace\checkstyle\src\xdocs-examples\resources\com\puppy
crawl\tools\checkstyle\checks\metrics\booleanexpressioncomplexity\Example2.java differ.
expected to be empty
but was: [25]
[ERROR]   BooleanExpressionComplexityCheckExamplesTest.testExample3:60->AbstractModuleTestSupport.verifyWithInlineConfigParser:238->Ab
stractModuleTestSupport.verifyViolations:470 Violation lines for E:\MANISH\Workspace\checkstyle\src\xdocs-examples\resources\com\puppy
crawl\tools\checkstyle\checks\metrics\booleanexpressioncomplexity\Example3.java differ.
expected to be empty
but was: [25]
[INFO]                                                                                                                                
[ERROR] Tests run: 3, Failures: 3, Errors: 0, Skipped: 0

```

am I missing something here ?

Reply:
please investigate why it works 
https://github.com/checkstyle/checkstyle/blob/101d1736d382047ea755d6cc8f54bda49e0e2624/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/annotation/annotationlocation/Example1.java#L31-L32

and we use this comments in regular Input, you can search over code to see more examples.

Reply:
> please investigate why it works


So, what seems to be happening is, the msg in `'....'`  either needs to be eactly the same as the `MSG_KEY` from `messages.properties`  or a simple `// violation above` comment. 

Doing this shouldn't be a problem for short MSG_KEY but most of them are lengthy leading to `LineLengthCheck` errors.
splitting violation comments inside ' ' into 2 lines is also not possible
If we try to customize msg or include partial msg, then we get the missing expected violation list as mentioned in comment above. 

I have been trying to find a workaround to this since long but nothing seems to strike yet  :)

> if you have // violation above you do not need // violation


I made changes in this PR (LineLength is almost close enough to 85 and works)